usr/lib/common/lock_sess_mgr.c

```diff
@@ -865,233 +917,338 @@ CK_RV session_mgr_set_op_state(STDLL_TokData_t *tokdata, SESSION *sess,
                                CK_OBJECT_HANDLE auth_key,
                                CK_BYTE *data, CK_ULONG data_len)
 {
+    CK_BYTE *cur_data;
+    CK_ULONG cur_data_len;
     OP_STATE_DATA *op_data = NULL;
     CK_BYTE *mech_param = NULL;
     CK_BYTE *context = NULL;
     CK_BYTE *ptr1 = NULL;
     CK_BYTE *ptr2 = NULL;
     CK_BYTE *ptr3 = NULL;
     CK_ULONG len;
+    CK_ULONG encr_key_needed = 0;
+    CK_ULONG auth_key_needed = 0;
 
     if (!sess || !data) {
         TRACE_ERROR("%s received bad argument(s)\n", __func__);
```

Comment:
Can it happen that data != NULL but data_len == 0?

Reply:
This is already checked in SC_SetOperationState.
And even if data_len would be zero, then the while loop below would just be skipped and the function would essentially do nothing.